File: HackMateBackend/HackMate/src/main/java/com/example/HackMateBackend/dtos/notification/UpdateNotificationPreferencesRequestDto.java
package com.example.HackMateBackend.dtos.notification;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UpdateNotificationPreferencesRequestDto {
    private boolean emailNotifications;
    private boolean pushNotifications;
    private boolean hackathonDeadlineReminders;
    private boolean joinRequestUpdates;
    private boolean teamUpdates;
    private boolean generalAnnouncements;
    private int reminderHoursBefore;
}

File: HackmateFrontendFolder/app/src/main/java/com/example/hackmatefrontendfolder/domain/model/notification/UpdateNotificationPreferencesRequestDto.kt
package com.example.hackmatefrontendfolder.domain.model.notification

data class UpdateNotificationPreferencesRequestDto(
    val emailNotifications : Boolean,
    val pushNotifications : Boolean,
    val hackathonDeadlineReminders : Boolean,
    val joinRequestUpdates : Boolean,
    val teamUpdates : Boolean,
    val generalAnnouncements : Boolean,
    val reminderHoursBefore : Int
)
